### ISSUE TITLE
Textfile reporter writing to arbitrary locations

### ISSUE THREAD

**Issue by PeterGottesman** [open-mpi/mtt#674]
When the python client is executing a section, if the section fails for some reason the results file will be written to the current working directory - i.e. wherever the failure occurred.
Additionally, it seems that when there is no tests to run it is just being written to $HOME.

**Comment by PeterGottesman** [comment]
```
~/
~/mttresults.txt <-- where it goes when nothing is done
~/PyMTT/ <-- Where I launch from
~/PyMTT/mttresults.txt <-- where it goes on a correct run
~/PyMTT/mtt/
~/scratch/
~/scratch/some_section/mttresults.txt <-- where it goes when it fails in some_section
```

**Comment by PeterGottesman** [comment]
Note that I am not sure pwd is the correct place to output it, so even on the "successful run" I think it is still not correct. I think it would be good to throw it in scratch unless there is a `--log-dir` or something similar.

**Comment by ribab** [comment]
Is this for the TextFile.py reporter plugin?

**Comment by PeterGottesman** [comment]
Ah, yes it is, I didn't catch that. That's what I get for not reading through the whole sample ini.
This should get picked up by #589 then?

**Comment by jsquyres** [comment]
Per discussion 2018-08-14, it might be easier to address this as writing to an absolute path name (vs. relying on what directory happens to be the cwd).

@ribab is going to address this in an upcoming sprint; he'll see if a rebased version of the branch he cited in #589 fixes this issue (and the issue cited in #589), and/or if it's worth converting this particular instance to write to an absolute path name.

Converting elsewhere in the code to use absolute filenames is likely a larger issue and outside the scope of this issue.
